include/buffer_control_block.h

```diff
@@ -22,8 +22,6 @@ typedef uint64_t File_Pointer;
 class BufferControlBlock {
 private:
   buffer_id_t id;
-  // busy lock
-  std::mutex mtx;
```

Comment:
I was afraid this was the issue. Is it possible to instead just enforce a global order on acquiring these mutexes? For instance, the canonical BFS order starting from root.

Reply:
For posterity: not the issue just removing dead code.